xcvm/lib/core/src/asset.rs

```diff
@@ -0,0 +1,215 @@
+use alloc::{collections::BTreeMap, string::ToString};
+use codec::{Decode, Encode};
+use fixed::{types::extra::U16, FixedU128};
+use scale_info::TypeInfo;
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+
+use crate::abstraction::IndexOf;
+
+/// Newtype for XCVM assets ID. Must be unique for each asset and must never change.
+/// This ID is an opaque, arbitrary type from the XCVM protocol and no assumption must be made on
+/// how it is computed.
+#[cfg_attr(feature = "std", derive(schemars::JsonSchema))]
+#[derive(
+	Copy,
+	Clone,
+	PartialEq,
+	Eq,
+	PartialOrd,
+	Ord,
+	Debug,
+	Encode,
+	Decode,
+	TypeInfo,
+	Serialize,
+	Deserialize,
+)]
+#[repr(transparent)]
+pub struct AssetID(pub u32);
```

Comment:
Asset id's are represented by `u128` in [XCVM spec](https://github.com/ComposableFi/composable/blob/main/xcvm/SPEC.md#asset-registries).

Reply:
Fixed